ui/src/app/core/capabilities.service.ts

```diff
@@ -0,0 +1,36 @@
+import {Headers, Http, RequestOptions} from '@angular/http';
+import {Injectable} from '@angular/core';
+import 'rxjs/add/operator/toPromise';
+
+import {CapabilitiesResponse} from '../shared/model/CapabilitiesResponse';
+import {environment} from '../../environments/environment';
+
+/** Service wrapper for accessing the capabilities endpoint. */
+@Injectable()
+export class CapabilitiesService {
+
+  private capabilitiesResponse: CapabilitiesResponse;
+
+  constructor(private http: Http) {}
+
+  getCapabilitiesSynchronous(): CapabilitiesResponse {
+    if (this.capabilitiesResponse) {
+      return this.capabilitiesResponse;
+    }
+    throw new Error("CapabilitiesResponse has not been retrieved yet.")
+  }
+
+  getCapabilities(): Promise<CapabilitiesResponse> {
+    if (this.capabilitiesResponse) {
```

Comment:
It would actually be good to still cache the promise here. In the event that two callers hit `getCapabilities()` on startup, we won't dispatch multiple requests.

Reply:
Done.